Source/WebKit/UIProcess/API/C/WKRemoteInspectorServer.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WKRemoteInspectorServer.h"
+
+#if ENABLE(REMOTE_INSPECTOR)
+#include <JavaScriptCore/RemoteInspectorServer.h>
+
+uint16_t WKRemoteInspectorServerStart(const char* address, uint16_t port)
+{
+    auto& server = Inspector::RemoteInspectorServer::singleton();
+    if (!server.start(address, port))
+        return 0;
+
+    return server.getPort().value_or(0);
+}
+#endif // ENABLE(REMOTE_INSPECTOR)
```

Comment:
An issue is that this will have a linker error if `!ENABLE(REMOTE_INSEPCTOR)`.
```suggestion
#if ENABLE(REMOTE_INSPECTOR)
#include <JavaScriptCore/RemoteInspectorServer.h>
#endif

uint16_t WKRemoteInspectorServerStart(const char* address, uint16_t port)
{
#if ENABLE(REMOTE_INSPECTOR)
    auto& server = Inspector::RemoteInspectorServer::singleton();
    if (!server.start(address, port))
        return 0;

    return server.getPort().value_or(0);
#else
    return 0;
#endif
}
```